include/oneapi/dpl/pstl/glue_memory_impl.h

```diff
@@ -303,22 +303,22 @@ template <class _ExecutionPolicy, class _ForwardIterator>
 oneapi::dpl::__internal::__enable_if_execution_policy<_ExecutionPolicy>
 uninitialized_value_construct(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardIterator __last)
 {
-    typedef typename ::std::iterator_traits<_ForwardIterator>::value_type _ValueType;
-    typedef ::std::decay_t<_ExecutionPolicy> _DecayedExecutionPolicy;
+    typedef typename std::iterator_traits<_ForwardIterator>::value_type _ValueType;
+    typedef std::decay_t<_ExecutionPolicy> _DecayedExecutionPolicy;
 
     auto __dispatch_tag = oneapi::dpl::__internal::__select_backend(__exec, __first);
 
-    if constexpr (::std::is_trivial_v<_ValueType>)
+    if constexpr (std::is_trivially_default_constructible_v<_ValueType> && std::is_trivially_copyable_v<_ValueType>)
```

Comment:
As in previous comments, it seems like `std::is_trivially_default_constructible_v<_ValueType>` is unnecessary, while `is_trivially_assignable` is necessary